Fix for Musixmatch multi-part lyrics

Ubiquitine opened the pull request:
## Description

Sometimes Musixmatch lyrics come in 2 parts.
Also, sometime non-existent lyrics return some content

## Changelog
### Lyrics: Improve Musixmatch backend
* Fix for lyrics that divided into 2 or more <div> elements.
* Ignore "fake" lyrics when lyrics not found but some content still being retrieved.

sampsyo commented on beetsplug/lyrics.py:
```diff
@@ -291,14 +291,25 @@ def fetch(self, artist, title):
+            lyrics_part1 = ''
+        html_part2 = html_parts[-1]
+        lyrics_part2 = extract_text_between(html_part2, '>', '</p>')
+        lyrics = lyrics_part1 + '\n' + lyrics_part2
```
Would it be simpler to do this with a loop over all the parts? Something roughly like this:

```py
for html_part in html_parts:
    lyrics_parts.append(extract_text_between(html_part1, '>', '</p>'))
lyrics = '\n'.join(lyrics_parts)
```

Ubiquitine replied:
Thanks for the correction.
Indeed, this approach is more elegant. I've updated my branch to use the loop, as well as better non-existing lyrics detection.
I've tested it on couple songs I have, and it seems to work.

sampsyo reviewed (commented):
Looks perfectly reasonable; thanks!! Here's a suggestion for how to simplify the code. Also, could you please add a quick changelog entry?

sampsyo commented:
Awesome; looks great! If you don't mind adding a quick changelog entry, I think this is good to go.

Ubiquitine commented:
I added changelog alongside the description. Or is there other place for this?

wisp3rwind commented:
> I added changelog alongside the description. Or is there other place for this?

I'm not seeing any changelog entry in this PR; did you actually push it? In any case, the changelog is at https://github.com/beetbox/beets/blob/master/docs/changelog.rst

Ubiquitine commented:
> > I added changelog alongside the description. Or is there other place for this?
> 
> I'm not seeing any changelog entry in this PR; did you actually push it? In any case, the changelog is at https://github.com/beetbox/beets/blob/master/docs/changelog.rst
Updated changelog.rst

sampsyo commented:
Looks great to me; thanks!!
